app/main.py

```diff
@@ -1,2 +1,61 @@
-class Dictionary:
-    pass
+from typing import Any
+
+
+class Node():
+    def __init__(self, key: Any, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
+class Dictionary():
+    def __init__(
+            self,
+            initial_capacity: int = 8,
+            load_factor: float = 0.75
+    ) -> None:
+        self.capacity = initial_capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.table = [None] * self.capacity
+
+    def _resize(self):
+        old_table = self.table
+        self.capacity *= 2
+        self.table = [None] * self.capacity
+        self.size = 0
+        
+        for node in old_table:
+            if node:
+                self.__setitem__(node.key, node.value)
+
+    def _index(self, key: Any) -> hash:
+        return hash(key) % self.capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.size / self.capacity > self.load_factor:
+            self._resize()
+
+        index = self._index(key)
+
+        while self.table[index] is not None:
+            if self.table[index].key == key:
+                self.table[index].value = value
+                return
+            index = (index + 1) % self.capacity
+
+        self.table[index] = Node(key, value)
+        self.size += 1
+
+    def __getitem__(self, key: Any) -> None:
```

Comment:
The return type of `__getitem__` should not be `None`. It should be `Any`, as it returns the value associated with the given key.